packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -780,7 +780,12 @@ class ScrollableState extends State<Scrollable> with TickerProviderStateMixin, R
         GestureBinding.instance.pointerSignalResolver.register(event, _handlePointerScroll);
       }
     } else if (event is PointerScrollInertiaCancelEvent) {
```

Comment:
OH! Or (sorry last thought) - Can this somehow be handled in the `pointerScroll` method instead?

Reply:
The more I think about this, the more I feel it is the right way to go. Can `PointerScrollInertiaCancelEvent` be a subclass of `PointerScrollEvent`?

Reply:
It seems like pointerScroll always goes idle after use. So it should work, even if not by design I guess.

Reply:
Oh never mind it is actually specified to go idle. So i'll try that instead

>   /// Any active animation is canceled. If the user is currently scrolling, that
>  /// action is canceled.

Reply:
I don't know if it is smart to remove this assertion, anyone who has their own implementation might have it as well, and get a breakage/assertion failure when inertia cancel is used. 

https://github.com/flutter/flutter/blob/e57b7f4ea8ab8b348810d0a76f7bcf4aeabbe6d2/packages/flutter/lib/src/widgets/scroll_position_with_single_context.dart#L210-L215

Reply:
That's ok. I think in the case of 0 we can just go idle and return.. ostensibly any change we make in the framework can end up breaking someone's custom widgets. I think it is worth investigating. We have methods in place to identify if it would be a big breaking change.